src/ts/core/features/filter-search.ts

```diff
@@ -0,0 +1,130 @@
+import {Feature, Settings} from '../settings'
+import {browser} from 'webextension-polyfill-ts'
+
+export const config: Feature = { // An object that describes new feature we introduce
+    id: 'filter_search',  // Feature id - any unique string would do
+    name: 'Filter Search',  // Feature name - would be displayed in the settings menu
+    //warning: 'Experimental feature',
+    enabledByDefault: true,
+}
+
+const checkSettingsAndSetupFilterSearchToggle = () => {
+    Settings.isActive('filter_search').then(active => {
+        toggleFilterSearch(active)
+    })
+}
+checkSettingsAndSetupFilterSearchToggle()
+
+browser.runtime.onMessage.addListener(async message => {
+    if (message === 'settings-updated') {
+        checkSettingsAndSetupFilterSearchToggle()
+    }
+})
+
+const toggleFilterSearch = (active: boolean) => {
+    const jsDoc = document;
+    if (active) {
+        jsDoc.addEventListener("click", filterSearch);
+    } else {
+        jsDoc.removeEventListener("click", filterSearch);
+    }
+}
+
+var debugMode = 0;
+
+function filterSearch(evt: MouseEvent)
+{
+    var evtTarget = (<HTMLElement>evt.target);
+    if(evtTarget !== null)
+    {
+        if(debugMode == 1){console.log(evtTarget.className);}
+
+        if(evtTarget.className === 'bp3-button')
+        {
+            var tbInputSearch = (<HTMLInputElement>document.getElementById("fbSearchInput"));
+            if(tbInputSearch !== null)
+            {
+                tbInputSearch.focus();
+                tbInputSearch.select();
+            }
+        }
+        else if(evtTarget.className === 'bp3-icon bp3-icon-filter' || evtTarget.className === 'bp3-button bp3-minimal bp3-small')
+        {
+            var waitCtr = 0;
+            function waitForFilter()
+            {
+                setTimeout(function(){
+                    //Check if filter search input box is there, otherwise need to load it
+                    var tbInputSearch = (<HTMLInputElement>document.getElementById("fbSearchInput"));
+
+                    if(tbInputSearch === null)
+                    {
+                        var allByClassTest = document.querySelectorAll('div.bp3-transition-container.bp3-popover-enter-done div.bp3-popover-content > div > div');
+                        var testFilterDiv = (<HTMLElement>allByClassTest[0]);
+                        if(typeof testFilterDiv !== 'undefined' && testFilterDiv !== null)
+                        {
+                            var newDivLine = document.createElement('div');
+                                newDivLine.className = 'rm-line';
```

Comment:
also it's an interesting way to format things, but I don't think it complies with the standard configured for the project. you can run `npm run prettify` to reformat things in a compliant way. you can also configure your editor to adhere to that (look for integration with `prettier`)